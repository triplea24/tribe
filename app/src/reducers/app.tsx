import { AnyAction } from "redux";

import { SHOW_LOADING, DISMISS_LOADING, SERVER_ERROR } from "../actions";

export interface State {
  loading: boolean;
  serverError: boolean;
}

export const initialState: State = {
  loading: false,
  serverError: false
};

export default (state: State = initialState, action: AnyAction) => {
  switch (action.type) {
    case SHOW_LOADING:
      return { ...state, loading: true };
    case DISMISS_LOADING:
      return { ...state, loading: false };
    case SERVER_ERROR:
      return { ...state, serverError: true };
  }
  return state;
};
