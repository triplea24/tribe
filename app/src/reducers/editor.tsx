import { AnyAction } from "redux";

import {
  CHANGE_EDITOR_MODE,
  CHANGE_BODY_CONTENT,
  CHANGE_TITLE,
  RESET_EDITOR,
  SHOW_EDITOR_ERROR,
  DISMISS_EDITOR_ERROR
} from "../actions";

const INITIAL_MODE = "article";

export interface State {
  mode: string;
  body: string;
  title: string;
  error: boolean;
  errorMessage: string;
}

export const initialState: State = {
  mode: INITIAL_MODE,
  body: "",
  title: "",
  error: true,
  errorMessage: ""
};

export default (state: State = initialState, action: AnyAction) => {
  switch (action.type) {
    case CHANGE_EDITOR_MODE:
      return { ...state, error: false, mode: action.payload };
    case CHANGE_BODY_CONTENT:
      return { ...state, error: false, body: action.payload };
    case CHANGE_TITLE:
      return { ...state, error: false, title: action.payload };
    case SHOW_EDITOR_ERROR:
      return { ...state, error: true, errorMessage: action.payload };
    case DISMISS_EDITOR_ERROR:
      return { ...state, error: false };
    case RESET_EDITOR:
      return { ...state, error: false, body: "", title: "" };
  }
  return state;
};
