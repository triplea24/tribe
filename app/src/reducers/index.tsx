import { omit, merge, assoc } from "ramda";

import {
  ADD_CONTENT,
  CHANGE_EDITOR_MODE,
  CHANGE_BODY_CONTENT,
  CHANGE_TITLE,
  RESET_EDITOR,
  REMOVE_CONTENT,
  LOAD_CONTENTS
} from "../actions";
import { AnyAction } from "redux";
export const initialState = {
  contents: {},
  editorMode: "article",
  bodyContent: "",
  title: ""
};

export interface State {
  contents: object;
  editorMode: string;
  bodyContent: string;
  title: string;
}

export default (state: State = initialState, action: AnyAction) => {
  switch (action.type) {
    case ADD_CONTENT:
      return {
        ...state,
        contents: assoc(action.payload._id, action.payload, state.contents)
      };
    case CHANGE_EDITOR_MODE:
      return { ...state, editorMode: action.payload };
    case CHANGE_BODY_CONTENT:
      return { ...state, bodyContent: action.payload };
    case CHANGE_TITLE:
      return { ...state, title: action.payload };
    case RESET_EDITOR:
      return { ...state, bodyContent: "", title: "" };
    case REMOVE_CONTENT:
      return { ...state, contents: omit([action.payload], state.contents) };
    case LOAD_CONTENTS:
      return { ...state, contents: merge(state.contents, action.payload) };
  }
  return state;
};
