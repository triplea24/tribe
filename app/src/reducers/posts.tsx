import { AnyAction } from "redux";
import { omit, merge, assoc } from "ramda";

import { REMOVE_CONTENT, LOAD_CONTENTS, ADD_CONTENT } from "../actions";

export interface State {
  data: object;
}
export const initialState: State = {
  data: {}
};

export default (state: State = initialState, action: AnyAction) => {
  switch (action.type) {
    case ADD_CONTENT:
      return {
        ...state,
        data: assoc(action.payload._id, action.payload, state.data)
      };
    case REMOVE_CONTENT:
      return { ...state, data: omit([action.payload], state.data) };
    case LOAD_CONTENTS:
      return { ...state, data: merge(state.data, action.payload) };
  }
  return state;
};
