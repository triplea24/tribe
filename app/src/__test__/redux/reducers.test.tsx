import configureStore from "redux-mock-store";
import ReduxThunk from "redux-thunk";

import reducers from "../../reducers";
import {
  CHANGE_EDITOR_MODE,
  CHANGE_BODY_CONTENT,
  CHANGE_TITLE,
  RESET_EDITOR
} from "../../actions";

const middlewares = [ReduxThunk];

const mockStore = configureStore(middlewares);
const initialState = {
  contents: {},
  editorMode: "article",
  bodyContent: "",
  title: ""
};

describe("Reducers", () => {
  describe("initial state", () => {
    const store = mockStore(initialState);
    it("should match a snapshot", () => {
      const initialState = store.getState();
      expect(initialState).toMatchSnapshot();
    });
  });

  describe("Editor mode", () => {
    const store = mockStore(initialState);
    it("it should change to post and vice versa", () => {
      const initialState: any = store.getState();
      let newState = reducers(initialState, {
        type: CHANGE_EDITOR_MODE,
        payload: "post"
      });
      expect(newState.editorMode).toBe("post");
      newState = reducers(initialState, {
        type: CHANGE_EDITOR_MODE,
        payload: "article"
      });
      expect(newState.editorMode).toBe("article");
    });
  });

  describe("Body content", () => {
    const store = mockStore(initialState);
    it("it should change body content", () => {
      const initialState: any = store.getState();
      const bodyContent = "Something";
      let newState = reducers(initialState, {
        type: CHANGE_BODY_CONTENT,
        payload: bodyContent
      });
      expect(newState.bodyContent).toBe(bodyContent);
      expect(newState).toMatchSnapshot();
    });
  });
  describe("Title", () => {
    const store = mockStore(initialState);
    it("it should change title", () => {
      const initialState: any = store.getState();
      const title = "Some cool title";
      let newState = reducers(initialState, {
        type: CHANGE_TITLE,
        payload: title
      });
      expect(newState.title).toBe(title);
      expect(newState).toMatchSnapshot();
    });
  });

  describe("Reset editor", () => {
    const store = mockStore(initialState);
    it("It should remove title and body", () => {
      const initialState: any = store.getState();
      const title = "Some cool title";
      const bodyContent = "Some cool boy content";
      let newState = reducers(initialState, {
        type: CHANGE_TITLE,
        payload: title
      });
      newState = reducers(newState, {
        type: CHANGE_BODY_CONTENT,
        payload: bodyContent
      });
      newState = reducers(newState, {
        type: RESET_EDITOR
      });
      expect(newState.title).toBe("");
      expect(newState.bodyContent).toBe("");
    });
  });
});
