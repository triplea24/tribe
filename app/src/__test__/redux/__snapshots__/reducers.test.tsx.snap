// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions Body content it should change body content 1`] = `
Object {
  "bodyContent": "Something",
  "contents": Object {},
  "editorMode": "article",
  "title": "",
}
`;

exports[`Actions Reset editor It should remove title and body 1`] = `
Object {
  "bodyContent": "",
  "contents": Object {},
  "editorMode": "article",
  "title": "Some cool title",
}
`;

exports[`Actions Reset editor it should change title 1`] = `
Object {
  "bodyContent": "",
  "contents": Object {},
  "editorMode": "article",
  "title": "Some cool title",
}
`;

exports[`Actions Title it should change title 1`] = `
Object {
  "bodyContent": "",
  "contents": Object {},
  "editorMode": "article",
  "title": "Some cool title",
}
`;

exports[`Actions initial state should match a snapshot 1`] = `
Object {
  "bodyContent": "",
  "contents": Object {},
  "editorMode": "article",
  "title": "",
}
`;
